lib/archive.rb

```diff
@@ -67,25 +67,26 @@ def self.sanitize_directories(files)
     starting_header = -1
 
     # add pre-existing directories to the set
-    for file in files
+    files.each do |file|
 
       # edge case for removing "./" from pathnames
+      # file[:pathname] = file[:pathname].gsub(/\.\.\//) {'..-'}
       if file[:pathname].include?("./")
-        file[:pathname] = file[:pathname].split("./")[1]
+        file[:pathname] = file[:pathname].split("./").last
```

Comment:
Actually, this approach is rather fragile and would break (not show the correct structure) on paths like `./b/./a/./a.txt` (multiple `./` in the middle)

![Screenshot 2024-04-15 at 13 54 04](https://github.com/autolab/Autolab/assets/9074856/3fba80e9-856e-4e5c-9eab-a2632e8e39dd)

What do you think about using something like `File.cleanpath(file[:pathname], rel_root = true).gsub("..", "__PARENT__")`?

`cleanpath` (https://www.rubydoc.info/gems/yard/File.cleanpath) will strip any `./`, and redundant `../` (e.g. `./b/../c` -> `c`)

The `gsub` would then help to indicate the correct relative hierarchy (for the remaining `../`):

![Screenshot 2024-04-15 at 13 51 32](https://github.com/autolab/Autolab/assets/9074856/b07227df-b32f-4d5d-a4ee-65a652c31046)

For testing, here are some files:
[test2.tar.zip](https://github.com/autolab/Autolab/files/14982686/test2.tar.zip)
**Contents**
```
./b/./a/./a.txt
```
[test.tar.zip](https://github.com/autolab/Autolab/files/14982691/test.tar.zip)
**Contents**
```
./a.txt
../b.txt
../../b/b.txt
../../c.txt
../../b/b.txt
../../../c/b/a/a.txt
```

Reply:
I didn't know `cleanpath` was a thing, it looks really good for this purpose!
I don't think this would be an accurate representation of the directory structure, since `a.txt` should be inside the __PARENT__ directories. However, I understand it might look confusing for the user if the file that should have been in the base directory was inside a nested directory. What do you think?

Reply:
Also, I think having the __PARENT__ directories might obstruct the view of the users from seeing the actual file outline clearly, considering the small width of the file tree.
![image](https://github.com/autolab/Autolab/assets/38949473/a7264280-a790-4ce8-8cb5-b743708ff521)

Reply:
Given that parent directories shouldn't occur that often, I don't think the obstructed view is a large concern.

I also think the behavior is fine regarding `a.txt` not being inside the `__PARENT__` directory. That seems to be the behavior for other software too (e.g. The Unarchiver)

![Screenshot 2024-04-20 at 12 49 29](https://github.com/autolab/Autolab/assets/9074856/d2d05224-3b1f-46aa-a470-a44cc55c2573)